insteon_mqtt/device/Dimmer.py

```diff
@@ -465,6 +465,69 @@ def set_backlight(self, level, on_done=None):
         msg_handler = handler.StandardCmd(msg, self.handle_backlight, on_done)
         self.send(msg, msg_handler)
 
+    #-----------------------------------------------------------------------
+    def get_flags(self, on_done=None):
+        """Hijack base get_flags to inject extended flags request.
+
+        The flags will be passed to the on_done callback as the data field.
+        Derived types may do something with the flags by override the
+        handle_flags method.
+
+        Args:
+          on_done: Finished callback.  This is called when the command has
+                   completed.  Signature is: on_done(success, msg, data)
+        """
+        seq = CommandSeq(self, "Dimmer get_flags complete", on_done,
+                         name="GetFlags")
+        seq.add(super().get_flags)
+        seq.add(self._get_ext_flags)
+        seq.run()
+
+    #-----------------------------------------------------------------------
+    def _get_ext_flags(self, on_done=None):
+        """Get the Insteon operational extended flags field from the device.
+
+        For the dimmer device, the flags include on-level and ramp-rate.
+
+        Args:
+          on_done: Finished callback.  This is called when the command has
+                   completed.  Signature is: on_done(success, msg, data)
+        """
+        LOG.info("Dimmer %s cmd: get extended operation flags", self.label)
+
+        # Requesting data d1 = 0x01
+        data = bytes([0x01] + [0x00] * 13)
+
+        msg = Msg.OutExtended.direct(self.addr, Msg.CmdType.EXTENDED_SET_GET,
+                                     0x00, data)
+        msg_handler = handler.ExtendedCmdResponse(msg, self.handle_ext_flags,
+                                                  on_done)
+        self.send(msg, msg_handler)
+
+    #-----------------------------------------------------------------------
+    def handle_ext_flags(self, msg, on_done):
+        """Handle replies to the _get_ext_flags command.
+
+        Extended message payload is:
+          D8 = on-level
+          D9 = ramp-rate
+
+        Args:
+          msg (message.InpExtended):  The message reply.
+          on_done:  Finished callback.  This is called when the command has
+                    completed.  Signature is: on_done(success, msg, data)
+        """
+        on_level = msg.data[7]
+        self.db.set_meta('on_level', on_level)
+        ramp_rate = msg.data[8]
+        for ramp_key, ramp_value in self.ramp_pretty.items():
+            if ramp_rate <= ramp_key:
+                ramp_rate = ramp_value
+                break
+        LOG.ui("Dimmer %s on_level: %s%% ramp rate: %ss", self.label,
+               on_level / 2.55, ramp_rate)
```

Comment:
Considering set_on_level takes a number in the range [0,255], maybe it would be better if this printed the raw value instead?  Or perhaps print both raw value + percentage?

(If you agree with this change, then the same should be made to insteon_mqtt/device/KeypadLinc.py.)

Reply:
It may also be worthwhile to limit the number of decimal places printed.  For example, I get the following with one of my dimmers:

> 2020-12-30 01:04:08 UI Dimmer: Dimmer 11.22.33 (kitchen cabinet lights) on_level: 70.19607843137256% ramp rate: 64s

Reply:
We define it in a scene using percentage, and I think for humans figuring out the brightness of 75% is easier than 191/255.  Plus it is what appears in home assistant.  I would actually say changing set_on_level to take a percentage would be better.

But agreed, this should be rounded to 1 or 2 decimal places.

Reply:
Interestingly enough, the brightness values that HA expects / provides by default are 0-255 ([HA documentation link](https://www.home-assistant.io/integrations/light.mqtt/#brightness_scale)) - they just get presented as a percentage in the UI.  Accordingly, the default [config.yaml](https://github.com/TD22057/insteon-mqtt/blob/68cfc6ca8f3c795d44edd3978d1cdb3077fc6978/config.yaml#L287) for dimmers expects to find / put 0-255 values in the MQTT topics.  As such, I think it's actually appropriate that set_flags / on_level takes a value in the same range, as it does now.  Furthermore, I'd be a bit concerned that we might introduce weird rounding errors by forcing these values to be converted to percentages for the MQTT interface and then back to raw numbers if we were to try to standardize on percentages.

As far as set_flags goes, I see that as a lower-level type of interface.  As such, I think people would not be particularly astonished to find that its interface smells a bit like the actual command being sent to the device.  On the other hand, I could also see some value in having a higher-level interface to get/set the on-level at some point - something like a pair of get/set default_on_level topics.  In that case, I think something that works a bit like the dimmer state_topic (which can do either 0-255 or percentages) and level_topic (which expects 0-255 values, but could probably be massaged with templating magic to take 0-100 if the user wanted it to) might make sense.

At any rate, we don't really need to solve this here/now.  I'll update to print both values for now.